eng/Versions.props

```diff
@@ -44,7 +44,7 @@
     <!-- Roslyn -->
     <MicrosoftCodeAnalysisVersion>3.3.1</MicrosoftCodeAnalysisVersion>
     <MicrosoftCodeAnalysisForRoslynDiagnosticsAnalyzersVersion>3.7.0</MicrosoftCodeAnalysisForRoslynDiagnosticsAnalyzersVersion>
-    <MicrosoftCodeAnalysisVersionForTests>4.0.1</MicrosoftCodeAnalysisVersionForTests>
+    <MicrosoftCodeAnalysisVersionForTests>4.2.0-1.22070.1</MicrosoftCodeAnalysisVersionForTests>
```

Comment:
❓   Was this necessary? It looks like parameterless struct constructors were merged to 4.0.1, in which case we would not need to update the test support as part of this pull request.

Reply:
When I tried to reproduce the issue, I had to bump the version but maybe in between something was merged. I will try to revert and will let you know (probably tonight, I am pretty full today).

Reply:
@sharwell So it wasn't required!